internal/backup/volumesnapshotcontent_action.go

```diff
@@ -64,6 +64,19 @@ func (p *VolumeSnapshotContentBackupItemAction) Execute(item runtime.Unstructure
 
 	// Create VolumeSnapshotBackup CR per VolumeSnapshotContent and add it as an additional item if its a DataMover case
 	if util.DataMoverCase() {
+
+		// check the VSC has the same backup name from label as the current backup
+		isVSForCurrentBackup, err := util.VSBHasVSBackupName(backup, &snapCont, p.Log)
+		if err != nil {
+			return nil, nil, errors.WithStack(err)
+		}
+
+		if !isVSForCurrentBackup {
```

Comment:
is this check redundant? aren't we returning err if the backup label doesnt match in https://github.com/openshift/velero-plugin-for-csi/pull/34/files#diff-74e34e12907f0dd88e604e4504da58b7d143d88e72336203200d7e95decaa4abR69-R72.